### PR TITLE
Feat/sni 4172 Ability to add applicant

### PR DESCRIPTION
**Pull request by ElifDenizEsea**
### Change description ###

Enter a description.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SNI-4172

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

**Review by ATedy — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/RecordNotifyParties.java [hunk 1]
```diff
@@ -47,6 +49,10 @@ public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseDa
 
         if (checkNullSubjectRepresentativeRespondent(data)) {
             errors.add("One recipient must be selected.");
+        } else if ((data.getCicCase().getCaseSubcategory() == CaseSubcategory.FATAL
+            || data.getCicCase().getCaseSubcategory() == CaseSubcategory.MINOR)
+            && !CollectionUtils.isEmpty(data.getCicCase().getNotifyPartySubject())) {
+            errors.add("Subject should not be selected for notification if the case is Fatal or Minor");
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/RecordNotifyParties.java:55]: Do we have some sort of Constants class where we can declare literals like this, as public static etc.?
As it's used in the class below, too.

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/ReinstateNotifyParties.java [hunk 2]
```diff
@@ -49,6 +51,10 @@ public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseDa
 
         if (checkNullSubjectRepresentativeRespondent(data)) {
             errors.add("One field must be selected.");
+        } else if ((data.getCicCase().getCaseSubcategory() == CaseSubcategory.FATAL
+            || data.getCicCase().getCaseSubcategory() == CaseSubcategory.MINOR)
+            && !CollectionUtils.isEmpty(data.getCicCase().getNotifyPartySubject())) {
+            errors.add("Subject should not be selected for notification if the case is Fatal or Minor");
         }
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/ReinstateNotifyParties.java:58]: Literal as above.

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/SelectRecipientsHelper.java [hunk 3]
```diff
@@ -52,6 +54,10 @@ public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseDa
 
         if (checkNullSubjectRepresentativeRespondent(data)) {
             errors.add("One recipient must be selected.");
+        } else if ((data.getCicCase().getCaseSubcategory() == CaseSubcategory.FATAL
+            || data.getCicCase().getCaseSubcategory() == CaseSubcategory.MINOR)
+            && !CollectionUtils.isEmpty(data.getCicCase().getNotifyPartySubject())) {
+            errors.add("Subject should not be selected for notification if the case is Fatal or Minor");
         }
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/SelectRecipientsHelper.java:61]: Literal...

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/RecordNotifyParties.java (hunk 1 above)
Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/RecordNotifyParties.java:55]: we have EventContants but it is mostly for event names and special characters.
I can move to there though, or create a new one. Up to you ?

**Review by osheaf01 — APPROVED** [review]
LGTM

**Review by ATedy — APPROVED** [review]

**Pull request closed without merging**